components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -57,6 +61,13 @@ class GeckoEngineSession(
 
     private var initialLoad = true
 
+    private val scope by lazy { CoroutineScope(Dispatchers.IO) }
```

Comment:
nit: Can you change other users of the history delegate to use this scope as well?

Reply:
Do you mean I should make this public and have `PlacesHistoryStorage` use this delegate? How would that work if `PlacesHistoryStorage` doesn't access the session; should I pass the scope as an argument to `GeckoEngineSession` and `PlacesHistoryStorage` instead?